src/lib/progress-spinner/progress-spinner.ts

```diff
@@ -7,62 +7,64 @@
  */
 
 import {
+  AfterViewInit,
   Component,
   ChangeDetectionStrategy,
-  OnDestroy,
   Input,
   ElementRef,
-  NgZone,
   Renderer2,
-  Directive,
-  ViewChild,
+  SimpleChanges,
+  OnChanges,
   ViewEncapsulation,
+  Optional,
+  Inject,
 } from '@angular/core';
 import {CanColor, mixinColor} from '@angular/material/core';
+import {Platform} from '@angular/cdk/platform';
+import {DOCUMENT} from '@angular/common';
 
-
-// TODO(josephperrott): Benchpress tests.
-
-/** A single degree in radians. */
-const DEGREE_IN_RADIANS = Math.PI / 180;
-/** Duration of the indeterminate animation. */
-const DURATION_INDETERMINATE = 667;
-/** Duration of the indeterminate animation. */
-const DURATION_DETERMINATE = 225;
-/** Start animation value of the indeterminate animation */
-const startIndeterminate = 3;
-/** End animation value of the indeterminate animation */
-const endIndeterminate = 80;
-/** Maximum angle for the arc. The angle can't be exactly 360, because the arc becomes hidden. */
-const MAX_ANGLE = 359.99 / 100;
-/** Whether the user's browser supports requestAnimationFrame. */
-const HAS_RAF = typeof requestAnimationFrame !== 'undefined';
-/** Default stroke width as a percentage of the viewBox. */
-export const PROGRESS_SPINNER_STROKE_WIDTH = 10;
-
+/** Possible mode for a progress spinner. */
 export type ProgressSpinnerMode = 'determinate' | 'indeterminate';
 
-type EasingFn = (currentTime: number, startValue: number,
-                 changeInValue: number, duration: number) => number;
-
-
-/**
- * Directive whose purpose is to add the mat- CSS styling to this selector.
- * @docs-private
- */
-@Directive({
-  selector: 'md-progress-spinner, mat-progress-spinner',
-  host: {'class': 'mat-progress-spinner'}
-})
-export class MdProgressSpinnerCssMatStyler {}
-
 // Boilerplate for applying mixins to MdProgressSpinner.
 /** @docs-private */
 export class MdProgressSpinnerBase {
   constructor(public _renderer: Renderer2, public _elementRef: ElementRef) {}
 }
 export const _MdProgressSpinnerMixinBase = mixinColor(MdProgressSpinnerBase, 'primary');
 
+/* tslint:disable:max-line-length */
+const INDETERMINATE_ANIMATION_TEMPLATE = `
+  .mat-progress-spinner-DIAMETER.mat-progress-spinner-indeterminate-animation[mode="indeterminate"] circle {
```

Comment:
Rather than constructing this selector, it might be better to just assign the `animation-name: mat-progress-spinner-stroke-rotate-DIAMETER` as an inline style. That way people can still override the timing and duration through CSS.

Reply:
Oh hey, that's a great idea. Should simplify this a lot.